### PR TITLE
[create_timepoint] upgraded module to react

### PR DESCRIPTION
**Pull request by maltheism**
## Brief summary of changes

This pull request removes smarty and upgrades module to use react. This PR removes the smarty templates and adds a react file for create_timepoint. Also it adds one php file for handling the post messages from the user.

Note: I changed how the visit label is handled. It's now done on the frontend and so there is no delay for the user.

#### Testing instructions (if applicable)

1. Pull the PR and test on your LORIS VM.

#### Links to related tickets (GitHub, Redmine, ...)

* Redmine? #16008

Please see #5238

### CODE REVIEW

**Comment by kongtiaowang** [comment]
@maltheism 
This PR has changed the process of create_timepoint. The test code need to be updated for matching the new REACT code. I am going to submit the test code in this PR.

**Comment by kongtiaowang** [comment]
it can't pass the manual test.

**Comment by johnsaigle** [comment]
@kongtiaowang Can you provide more details about the bugs you encountered while testing?

**Comment by kongtiaowang** [comment]
@johnsaigle I told @maltheism, and she will update this PR.

**Comment by maltheism** [comment]
@johnsaigle @kongtiaowang I just fixed it.

**Comment by kongtiaowang** [comment]
need to fix phan error, 
modules/create_timepoint/php/endpoints/timepoint.class.inc:202 PhanUndeclaredVariable Variable $project is undeclared (Did you mean $projects)

File: test/run-php-linter.sh [hunk 1]
```diff
@@ -17,5 +17,3 @@ vendor/bin/phpcs --standard=docs/LorisCS.xml --extensions=php/php,inc/php module
 
 # Run PHPCS on src/ directory using a different ruleset conforming to PSR2.
 vendor/bin/phpcs --standard=docs/SrcCS.xml --extensions=php/php src/ || exit $?;
-
-vendor/bin/phpmd php/libraries text docs/LorisPHPMD.xml || exit $?;
```
Comment (johnsaigle) [inline test/run-php-linter.sh]: This shouldn't be removed in this PR.

Comment (maltheism) [reply test/run-php-linter.sh]: Hey @johnsaigle, this is marked as Outdated and you made the comment 34 minutes ago after the commit that resolved this. Is there something still wrong with it? The file looks the same as major.

Comment (johnsaigle) [reply test/run-php-linter.sh]: Should be fine

File: modules/create_timepoint/php/create_timepoint.class.inc [hunk 2]
```diff
@@ -48,256 +35,28 @@ class Create_Timepoint extends \NDB_Form
...
-
         // check user permissions
         return (
             $user->hasPermission('data_entry') &&
             (in_array(
-                $candidate->getData('RegistrationCenterID'),
-                $user->getData('CenterIDs')
+                intval($candidate->getData('RegistrationCenterID')),
+                $user->getData('CenterIDs'),
+                true
```
Comment (johnsaigle) [inline modules/create_timepoint/php/create_timepoint.class.inc]: These lines would be better as:

```php
return $user->hasPermission('data_entry') 
&&  $user->hasCenter($candidate->getCenterID());
```

**Comment by maltheism** [comment]
Closing in favour of https://github.com/aces/Loris/pull/5831. I ran into difficulties rebasing to master.

**Pull request closed without merging**